[BUILD] Transitive dependency issue with the otlp http exporter

**Describe your environment** 
Ubuntu latest and Fedora 38

**Steps to reproduce**
Build main (or 1.9.0) with cmake with -DWITH_OTLP=on -DWITH_OTLP_HTTP=on -DWITH_OTLP_GRPC=off either with static or shared libraries (I'm really mostly interested in shared libs though).

run 'make' then 'make install' installing in a place that the subsequent qpid-proton build will find.

Then configure qpid-proton with the appropriate install prefix

**What is the expected behavior?**
No error!

**What is the actual behavior?**
...
CMake Error at cpp/examples/CMakeLists.txt:62 (add_executable):
  Target "tracing_client" links to target "protobuf::libprotobuf" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at cpp/examples/CMakeLists.txt:62 (add_executable):
  Target "tracing_client" links to target "nlohmann_json::nlohmann_json" but
  the target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at cpp/examples/CMakeLists.txt:62 (add_executable):
  Target "tracing_server" links to target "protobuf::libprotobuf" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?


CMake Error at cpp/examples/CMakeLists.txt:62 (add_executable):
  Target "tracing_server" links to target "nlohmann_json::nlohmann_json" but
  the target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?
...

**Additional context**
Here is a Dockerfile that builds under ubuntu latest demonstrating the problem:
[Dockerfile.txt](https://github.com/open-telemetry/opentelemetry-cpp/files/11547303/Dockerfile.txt)

Looking at the installed exporter header files for the otlp http exporter there is no actual reference to either nlohmann_json headers or protobuf headersand so I can see no reason why ot defines them as public dependencies (or even why they would be interface deps).

--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -113,8 +113,9 @@ if(WITH_OTLP_HTTP)
 
   target_link_libraries(
     opentelemetry_exporter_otlp_http_client
-    PUBLIC opentelemetry_sdk opentelemetry_proto opentelemetry_http_client_curl
-           nlohmann_json::nlohmann_json)
+    PUBLIC opentelemetry_sdk
+    PRIVATE opentelemetry_proto opentelemetry_http_client_curl
+            nlohmann_json::nlohmann_json)
   if(nlohmann_json_clone)
     add_dependencies(opentelemetry_exporter_otlp_http_client
                      nlohmann_json::nlohmann_json)
@@ -197,8 +198,9 @@ endif()
 
 if(BUILD_TESTING)
   add_executable(otlp_recordable_test test/otlp_recordable_test.cc)
-  target_link_libraries(otlp_recordable_test ${GTEST_BOTH_LIBRARIES}
-                        ${CMAKE_THREAD_LIBS_INIT} opentelemetry_otlp_recordable)
+  target_link_libraries(
+    otlp_recordable_test ${GTEST_BOTH_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
+    opentelemetry_otlp_recordable protobuf::libprotobuf)
   gtest_add_tests(
     TARGET otlp_recordable_test
     TEST_PREFIX exporter.otlp.
@@ -217,8 +219,10 @@ if(BUILD_TESTING)
 
   add_executable(otlp_metrics_serialization_test
                  test/otlp_metrics_serialization_test.cc)
-  target_link_libraries(otlp_metrics_serialization_test ${GTEST_BOTH_LIBRARIES}
-                        ${CMAKE_THREAD_LIBS_INIT} opentelemetry_otlp_recordable)
+  target_link_libraries(
+    otlp_metrics_serialization_test ${GTEST_BOTH_LIBRARIES}
+    ${CMAKE_THREAD_LIBS_INIT} opentelemetry_otlp_recordable
+    protobuf::libprotobuf)
   gtest_add_tests(
     TARGET otlp_metrics_serialization_test
     TEST_PREFIX exporter.otlp.
@@ -307,9 +311,14 @@ if(BUILD_TESTING)
   if(WITH_OTLP_HTTP)
     add_executable(otlp_http_exporter_test test/otlp_http_exporter_test.cc)
     target_link_libraries(
-      otlp_http_exporter_test ${GTEST_BOTH_LIBRARIES} ${CMAKE_THREAD_LIBS_INIT}
-      ${GMOCK_LIB} opentelemetry_exporter_otlp_http
-      opentelemetry_http_client_nosend)
+      otlp_http_exporter_test
+      ${GTEST_BOTH_LIBRARIES}
+      ${CMAKE_THREAD_LIBS_INIT}
+      ${GMOCK_LIB}
+      opentelemetry_exporter_otlp_http
+      opentelemetry_http_client_nosend
+      nlohmann_json::nlohmann_json
+      protobuf::libprotobuf)
     gtest_add_tests(
       TARGET otlp_http_exporter_test
       TEST_PREFIX exporter.otlp.
@@ -362,7 +371,9 @@ if(BUILD_TESTING)
       ${GMOCK_LIB}
       opentelemetry_exporter_otlp_http_metric
       opentelemetry_metrics
-      opentelemetry_http_client_nosend)
+      opentelemetry_http_client_nosend
+      nlohmann_json::nlohmann_json
+      protobuf::libprotobuf)
     gtest_add_tests(
       TARGET otlp_http_metric_exporter_test
       TEST_PREFIX exporter.otlp.

--- a/exporters/otlp/CMakeLists.txt
+++ b/exporters/otlp/CMakeLists.txt
@@ -113,7 +113,7 @@ if(WITH_OTLP_HTTP)
 
   target_link_libraries(
     opentelemetry_exporter_otlp_http_client
-    PUBLIC opentelemetry_sdk
+    PUBLIC opentelemetry_sdk opentelemetry_ext
     PRIVATE opentelemetry_proto opentelemetry_http_client_curl
             nlohmann_json::nlohmann_json)
   if(nlohmann_json_clone)

